java/src/org/openqa/selenium/remote/service/DriverFinder.java

```diff
@@ -0,0 +1,40 @@
+package org.openqa.selenium.remote.service;
+
+import org.openqa.selenium.Capabilities;
+import org.openqa.selenium.internal.Require;
+import org.openqa.selenium.manager.SeleniumManager;
+import org.openqa.selenium.os.ExecutableFinder;
+
+import java.io.File;
+import java.util.logging.Logger;
+
+public class DriverFinder {
+  private static final Logger LOG = Logger.getLogger(DriverFinder.class.getName());
+
+  public static String getPath(Capabilities options, String driverName, String driverProperty) {
+    String exePath = new ExecutableFinder().find(driverName);
+    if (exePath == null) {
+      exePath = System.getProperty(driverProperty);
+    }
+    if (exePath == null) {
+      try {
+        exePath = SeleniumManager.getInstance().getDriverPath(options);
```

Comment:
Maybe we can reduce this to 

```
exePath = System.getProperty(driverProperty, SeleniumManager.getInstance().getDriverPath(options));
```

Reply:
Soon the Selenium manager will be handling the path logic as well... And soon we'll want to output the Selenium manager error instead of catching/hiding. So this code will get optimized again soon, but want to figure that out separately.